pkg/build/controller/strategy/docker.go

```diff
@@ -17,7 +16,7 @@ type DockerBuildStrategy struct {
 // CreateBuildPod creates the pod to be used for the Docker build
 // TODO: Make the Pod definition configurable
 func (bs *DockerBuildStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.Pod, error) {
-	buildJSON, err := json.Marshal(build)
+	data, err := v1beta1.Codec.Encode(build)
```

Comment:
Inject the codec from the BuildControllerFactory through the typeBasedFactoryStrategy.